app/models/dob.rb

```diff
@@ -0,0 +1,74 @@
+class Dob < Base
+
+  attribute :date_of_birth, :datetime
+  
+  validates :date_of_birth, presence: true
+  
+  def next_step
+   "uk_or_overseas" 
+  end
+
+  def initialize(attributes = nil)
+    super
+
+    assign_attributes(attributes) if attributes
+  end
+
+  def assign_attributes(attributes)
+    new_attributes = attributes.stringify_keys
+    multiparameter_attributes = extract_multiparameter_attributes(new_attributes)
+
+    multiparameter_attributes.each do |multiparameter_attribute, values_hash|
+      set_values = (1..3).collect{ |position| values_hash[position].to_i }
+   
+    if set_values.include?(0)
```

Comment:
Could these not be custom validators and applied next to the presence validator?

Reply:
See below.

Reply:
The errors referenced in the if.. block don't get used, really you can just return from the assign_attributes method, this assigns nil effectively to the dob instance, so the presence check will fail. Not the best solution admittedly.